## Practice Exercise 1

You are provided with 2 lists that contain the data of an ecommerce website. The first list contains the data for the number of items sold for a particular product and the second list contains the price of the product sold. As a part of this exercise, solve the questions that are provided below.

In [2]:
number = [8, 9, 9, 1, 6, 9, 5, 7, 3, 9, 7, 3, 4, 8, 3, 5, 8, 4, 8, 7, 5, 7, 3, 6, 1, 2, 7, 4, 7, 7, 8, 4, 3, 4, 2, 2, 2, 7, 3, 5, 6, 1, 1, 3, 2, 1, 1, 7, 7, 1, 4, 4, 5, 6, 1, 2, 7, 4, 5, 8, 1, 4, 8, 6, 2, 4, 3, 7, 3, 6, 2, 3, 3, 3, 2, 4, 6, 8, 9, 3, 9, 3, 1, 8, 6, 6, 3, 3, 9, 4, 6, 4, 9, 6, 7, 1, 2, 8, 7, 8, 1, 4]
price = [195, 225, 150, 150, 90, 60, 75, 255, 270, 225, 135, 195, 30, 15, 210, 105, 15, 30, 180, 60, 165, 60, 45, 225, 180, 90, 30, 210, 150, 15, 270, 60, 210, 180, 60, 225, 150, 150, 120, 195, 75, 240, 60, 45, 30, 180, 240, 285, 135, 165, 180, 240, 60, 105, 165, 240, 120, 45, 120, 165, 285, 225, 90, 105, 225, 45, 45, 45, 75, 180, 90, 240, 30, 30, 60, 135, 180, 15, 255, 180, 270, 135, 105, 135, 210, 180, 135, 195, 225, 75, 225, 15, 240, 60, 15, 180, 255, 90, 15, 150, 230, 150]
import numpy as np
new_number=np.array(number)

new_number.size

102

#### How many different products are sold by the company in total?

-  99
-  100
-  101
-  102


In [3]:
# Type your code here
np.sum(new_number)

490

#### How many items were sold in total?

-  460
-  490
-  500
-  520


In [15]:
# Type your code here
new_price=np.array(price)
aver=np.average(new_price)
aver

139.01960784313727

#### What is the average price of the products sold by the ecommerce company?

-  139
-  151
-  142
-  128

In [9]:
# Type your code here
max(new_price)

285

#### What is the price of the costliest item sold?

-  225
-  310
-  280
-  285

In [10]:
# Type your code here
np.sum(new_price*new_number)

67100

#### What is the total revenue of the company? [Revenue = Price\*Quantity]

-  67100
-  53900
-  45300
-  71200

In [13]:
# Type your code here
new_number_1=(new_number[20]>new_number[50])
new_number_1

True

#### Demand for the 20th product in the list is more than the 50th product. [True/False]
-  True
-  False
-  Can't be calculated

In [24]:
# Type your code here
exp=new_price[new_price>aver]
exp.size

52

#### How many products fall under the category of expensive goods? 
An expensive good is that good whose price is more than the average price of the products sold by the company.

-  48
-  50
-  52
-  54

In [ ]:
# Type your code here

In [25]:
x=[23,82]
print((''.join(list(map(str,x)))),)

2382


In [28]:
t=[(2,1),(1,1),(2,4),(2,6),(7,5),(6,4)]
result={}
for i,j in t:
    result.setdefault(j,[]).append(i)
print(result)

{1: [2, 1], 4: [2, 6], 6: [2], 5: [7]}


In [30]:
A=np.array([[1,2,3],[4,5,6],[7,8,9]])
B=np.array([[9,8,7],[6,5,4],[3,2,1]])
M=A>=B
print(M*A)

[[0 0 0]
 [0 5 6]
 [7 8 9]]


In [46]:
import json
sampleJson="""{
    "Partner":{
        "Merchant":{
            "name":"Merhcant1",
            "Dimensions":{
            "Product":"Paypal",
            "Country:"US,
            "Date of Join":"2019-09-01"
            }
        }
    }
}"""

In [50]:
data=json.loads(sampleJson)

print(data['Partner.name.Dimensions' = "Merhcant1",'Date of Join'])

#print(data['Partner']['name']['Dimensions']['Date of Join'])

SyntaxError: invalid syntax (<ipython-input-50-9cfad186c035>, line 3)

In [40]:
def sample(args):
    args.insert(args.index(args[-1]),2)
    args.pop(0)
    
numbers=replica=[3,4,5,6]
sample(replica)
print(replica)

[4, 5, 2, 6]
